inginious/frontend/pages/admin/admin.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INGInious. See the LICENSE and the COPYRIGHTS files for
+# more information about the licensing of this file.
+
+""" Admin index page"""
+from itertools import islice
+
+from flask import redirect, request, jsonify
+
+from inginious.frontend.pages.utils import INGIniousAdministratorPage
+
+
+class AdministrationUsersPage(INGIniousAdministratorPage):
+    """User Admin page."""
+
+    def GET_AUTH(self, *args, **kwargs):
+        """ Display admin users page """
+        return self.show_page()
+
+    def POST_AUTH(self, *args, **kwargs):
+        """ Display admin users page"""
+        return self.show_page()
+
+    def show_page(self):
+        """Display page"""
+
+        def _chunks(data, size):
+            _it = iter(data)
+            for _ in range(0, len(data), size):
+                yield {k: data[k] for k in islice(_it, size)}
+
+        page = int(request.form.get("page")) if request.form.get("page") is not None else 1
+
+        all_users = self.user_manager.get_users()
+        size_users = len(all_users)
+        user_per_page = 10
+        pages = size_users // user_per_page + (size_users % user_per_page > 0)
+        subdicts = list(_chunks(all_users, user_per_page))
+        display_users = subdicts[page - 1]
```

Comment:
Although it seems to perform not so bad with thousands of users and a local database, you should not fetch all users, parse the whole set of data and then throw everything but 10 results.